lib/maths/CBoostedTreeImpl.cc

```diff
@@ -686,28 +685,53 @@ CBoostedTreeImpl::trainForest(core::CDataFrame& frame,
             this->refreshPredictionsAndLossDerivatives(frame, trainingRowMask, eta, tree);
             forest.push_back(std::move(tree));
             eta = std::min(1.0, m_EtaGrowthRatePerTree * eta);
-            oneMinusBias += eta * (1.0 - oneMinusBias);
             retries = 0;
-        } else if (oneMinusBias < 1.0) {
-            scopeMemoryUsage.add(tree);
-            this->refreshPredictionsAndLossDerivatives(frame, trainingRowMask, 1.0, tree);
-            oneMinusBias = 1.0;
-            forest.push_back(std::move(tree));
+            m_TrainingProgress.increment();
+        } else {
+            candidateSplits = this->candidateSplits(frame, downsampledRowMask);
+            nextTreeCountToRefreshSplits +=
+                static_cast<std::size_t>(std::max(0.5 / eta, 2.0));
```

Comment:
It would be nice to have a comment explaining what is going on here.

Reply:
I added a catch all comment for amortising cost computing splits. See [56f32ea](https://github.com/elastic/ml-cpp/pull/811/commits/56f32eae2a4f4baa7867dfbd164c389022288cfe). I didn't explicitly mention why I tied to eta, i.e. captures something about how different we expect splits chosen from round-to-round to be. Does this seem sufficient to you?